drivers/mpu_9250.cpp

```diff
@@ -0,0 +1,514 @@
+
+    // Use slv0 to access mag raw datas (x,y,z and status 2 register)
+    // burst read used
+    uint8_t first_reg       = AK8963_HXL;
+    uint8_t address         = (uint8_t)(MPU9250_AK8963_ADDR | MPU9250_READ_FLAG);
+    uint8_t slv0_ctrl_reg   = (uint8_t)(MPU9250_I2CSLV_EN | 7);
+
+    success &= write_reg(MPU9250_SLV0_REG_REG , &first_reg);
+    success &= write_reg(MPU9250_SLV0_ADDR_REG, &address);
+    success &= write_reg(MPU9250_SLV0_CTRL_REG, &slv0_ctrl_reg);
+
+    // Configuring Accelerometer and Gyroscope
+    success &= set_acc_lpf();
+    success &= set_gyro_lpf();
+    success &= set_mpu_sample_rate();
+    success &= set_acc_range();
+    success &= set_gyro_range();
+
+    return success;
+}
+
+
+bool Mpu_9250::update_acc(void)
+{
+    bool success = true;
+
+    // Read raw data from accelerometer (big endian)
+    uint8_t acc_data_raw[6] = {0};
+    success &= read_reg(MPU9250_ACCEL_X_OUT_MSB, acc_data_raw, 6);
+
+    acc_data_[0] = (float)((int16_t)(acc_data_raw[0] << 8 | acc_data_raw[1]));
+    acc_data_[1] = (float)((int16_t)(acc_data_raw[2] << 8 | acc_data_raw[3]));
+    acc_data_[2] = (float)((int16_t)(acc_data_raw[4] << 8 | acc_data_raw[5]));
+
+    acc_data_[0] /= acc_scale_;
+    acc_data_[1] /= acc_scale_;
+    acc_data_[2] /= acc_scale_;
+
+    last_update_us_ = time_keeper_get_us();
+
+    return success;
+}
+
+
+bool Mpu_9250::update_gyr(void)
+{
+    bool success = true;
+
+    // Read raw data from gyroscope (big endian)
+    uint8_t gyro_data_raw[6] = {0};
+    success &= read_reg(MPU9250_GYRO_X_OUT_MSB, gyro_data_raw, 6);
+
+    gyro_data_[0] = (float)((int16_t)(gyro_data_raw[0] << 8 | gyro_data_raw[1]));
+    gyro_data_[1] = (float)((int16_t)(gyro_data_raw[2] << 8 | gyro_data_raw[3]));
+    gyro_data_[2] = (float)((int16_t)(gyro_data_raw[4] << 8 | gyro_data_raw[5]));
+
+    // Gyroscope is sending dps, scaling it to get rps
+    gyro_data_[0] /= gyro_scale_;
+    gyro_data_[1] /= gyro_scale_;
+    gyro_data_[2] /= gyro_scale_;
+    
+    last_update_us_ = time_keeper_get_us();
+
+    return success;
+}
+
+bool Mpu_9250::update_mag(void)
+{
+    bool success = true;
+
+    // Read raw data from magnetometer (little endian)
+    uint8_t mag_data_raw[6] = {0};
+    success &= read_reg(MPU9250_EXT_SENS_DATA_00, mag_data_raw, 6);
+    
+    mag_data_[0] = (float)((int16_t)(mag_data_raw[1] << 8 | mag_data_raw[0]));
+    mag_data_[1] = (float)((int16_t)(mag_data_raw[3] << 8 | mag_data_raw[2]));
+    mag_data_[2] = (float)((int16_t)(mag_data_raw[5] << 8 | mag_data_raw[4]));
+
+    // mag_data_[0] /= mag_scale_;
+    // mag_data_[1] /= mag_scale_;
+    // mag_data_[2] /= mag_scale_;
+    
+    last_update_us_ = time_keeper_get_us();
+
+    return success;
+}
+
+
+const float& Mpu_9250::last_update_us(void) const
+{
+    return last_update_us_;
+}
+
+
+const std::array<float, 3>& Mpu_9250::gyro(void) const
+{
+    return gyro_data_;
+}
+
+
+const float& Mpu_9250::gyro_X(void) const
+{
+    return gyro_data_[0];
```

Comment:
@jlecoeur 
We used to have a define for the axis order before
If we want to put the board in a different orientation for example
How this would work in this case ?